src/redux/books/booksSlice.js

```diff
@@ -1,73 +1,40 @@
-import { createSlice, createAsyncThunk } from '@reduxjs/toolkit';
-import axios from 'axios';
-// API URL
-const booksAPI = 'https://us-central1-bookstore-api-e63c8.cloudfunctions.net/bookstoreApi/apps/UUD89QFWkJvMJew3gPgt';
-
-// Async thunk for adding a book
-export const addBook = createAsyncThunk('books/addBook', async (book) => {
-  try {
-    await axios.post(`${booksAPI}/books`, book);
-  } catch (e) {
-    throw new Error(e);
-  }
-  return book;
-});
-
-export const deleteBook = createAsyncThunk('books/deleteBook', async (bookId) => {
-  try {
-    await axios.delete(`${booksAPI}/books/${bookId}`);
-  } catch (e) {
-    throw new Error(e);
-  }
-  return bookId;
-});
-
-export const getBooks = createAsyncThunk('books/getBooks', async () => {
-  const response = await axios.get(`${booksAPI}/books`);
-
-  const books = Object.entries(response.data).map((item) => (
-    {
-      ...item[1][0],
-      item_id: item[0],
-    }
-  ));
-  return books;
-});
+import { createSlice } from '@reduxjs/toolkit';
 
 const initialState = {
-  books: [],
-  isLoading: false,
-  isError: undefined,
+  books: [
+    {
+      itemId: 'item1',
+      title: 'The Great Gatsby',
+      author: 'John Smith',
+      category: 'Fiction',
+    },
+    {
+      itemId: 'item2',
+      title: 'Anna Karenina',
+      author: 'Leo Tolstoy',
+      category: 'Fiction',
+    },
+    {
+      itemId: 'item3',
+      title: 'The Selfish Gene',
+      author: 'Richard Dawkins',
+      category: 'Nonfiction',
+    },
+  ],
 };
 
 const booksSlice = createSlice({
   name: 'books',
   initialState,
-  reducers: {},
-
-  extraReducers: (builder) => {
-    builder.addCase(addBook.fulfilled, (state, action) => {
+  reducers: {
+    removeBook: (state, action) => {
+      state.books = state.books.filter((book) => book.itemId !== action.payload);
+    },
+    addBook: (state, action) => {
       state.books.push(action.payload);
-    });
```

Comment:
- [ ] There should be a `fetchBooks` action that fetches the books from the API when a user loads the page. Kindly add this feature so that when a user loads the page, the books are fetched and displayed on the browser.

- [ ] Your app does not work as required. Books should be added to API when a user submits the form, and the user should be able to remove the book from the API when they click the remove button.

- [ ] These three actions should be added using `createAsyncThunk` and `axios`/`fetch`

Reply:
I have done the changes you suggested. Thanks a lot for your positive feedback. Regards